### ISSUE TITLE
:sparkles: Can we have text style widgets back?

### ISSUE THREAD

**Issue by L0bsang** [jurialmunkey/skin.arctic.fuse#717]
### Skin section

Widgets

### Feature description

I would like a return of the text style for widgets, mostly for use with displaying trakt lists in hubs. I imagine there is a plan to return this style at some point, just confirming this is the case.

### Why is this feature needed? Please justify and explain.

We used to have a pure text style widget, was recently removed I believe in this commit: 
https://github.com/jurialmunkey/skin.arctic.fuse/commit/e7e9be6af63c4e5d978fb51ec857e613f6399eac 
If I understand correctly, that change was done to standardize the widget types (icon is now smaller square for example). 

There is a lot of use for this type of widget (if it conforms to the rest of the skin's style and standards. 
I know we can use submenus, but when the contents of that submenu's items is a list of items (for example trakt lists) all with the same icon; it is then nice to have a widget that is just the list titles, then another stacked widget below to show the actual content. I would also accept stacked submenus. 

### Checklist

- [X] I have made a separate issue for each feature I am requesting.
- [X] I have searched the issues and this feature has not been requested.
- [X] I am using the most recent skin version.

**Comment by jurialmunkey** [comment]
I've got a bit of a different plan for the "Text" style but it works quite well for Trakt Lists:

![screenshot00514](https://github.com/jurialmunkey/skin.arctic.fuse/assets/4618719/c52979d4-24ac-450c-8aaa-30c8d8af9beb)


There's also a larger "Overview" equivalent style:

![screenshot00516](https://github.com/jurialmunkey/skin.arctic.fuse/assets/4618719/cd263016-cab9-421a-a06a-00105a2ff7f6)

**Comment by jurialmunkey** [comment]
Just trying to decide whether I prefer the title text outside or inside the box. I think maybe outside is better. Plus makes it a little more consistent with other styles

![screenshot00523](https://github.com/jurialmunkey/skin.arctic.fuse/assets/4618719/67dfa621-9633-44f7-96fd-9792b154ff64)

![screenshot00524](https://github.com/jurialmunkey/skin.arctic.fuse/assets/4618719/e1da319a-cc47-4873-a1ee-b01e7c6d8beb)

**Comment by CyberRooke** [comment]
> Just trying to decide whether I prefer the title text outside or inside the box. I think maybe outside is better. Plus makes it a little more consistent with other styles
> 
> ![screenshot00523](https://github.com/jurialmunkey/skin.arctic.fuse/assets/4618719/67dfa621-9633-44f7-96fd-9792b154ff64)
> 
> ![screenshot00524](https://github.com/jurialmunkey/skin.arctic.fuse/assets/4618719/e1da319a-cc47-4873-a1ee-b01e7c6d8beb)
> 

I think the inside may be better... gives it a kind of "full package" look, whereas the outside makes the title and genre look isolated and could lead to users asking for a way to switch the genre information off 😅... 

The inside look is also more visually appealing, in my opinion

**Comment by Liltune95** [comment]
> Just trying to decide whether I prefer the title text outside or inside the box. I think maybe outside is better. Plus makes it a little more consistent with other styles
> 
> ![screenshot00523](https://private-user-images.githubusercontent.com/4618719/321478143-67dfa621-9633-44f7-96fd-9792b154ff64.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4MTA3ODksIm5iZiI6MTcxMjgxMDQ4OSwicGF0aCI6Ii80NjE4NzE5LzMyMTQ3ODE0My02N2RmYTYyMS05NjMzLTQ0ZjctOTZmZC05NzkyYjE1NGZmNjQuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTFUMDQ0MTI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmZhNDY1ZTRlYzM3YThlZmM4N2Q3NzUwNTA5ZTQyYTgzZTA2OGUwNTc0M2JkMWE4Y2Q1MTlhZWIwYTVkZjZmZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.zhpjRr2o8r_uyl5acH_Wm9-KPnFCBrGIZCIcF2cpgcc)
> 
> ![screenshot00524](https://private-user-images.githubusercontent.com/4618719/321478150-e1da319a-cc47-4873-a1ee-b01e7c6d8beb.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4MTA3ODksIm5iZiI6MTcxMjgxMDQ4OSwicGF0aCI6Ii80NjE4NzE5LzMyMTQ3ODE1MC1lMWRhMzE5YS1jYzQ3LTQ4NzMtYTFlZS1iMDFlN2M2ZDhiZWIuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTFUMDQ0MTI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDU1ZGMwZGFhMzg4MjhmN2FlNWI4MmY2Yzc4NjU2ODc5NzhmODZjYTZlZjEwYzVkOWJlM2Q2Mzc1ZGQ5YTIzZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.uIzDW0Cy-0HzuEZbPcKXkFsOKY7eJPnRr1U850FW4nA)

Def inside text, Looks more "complete".

**Comment by L0bsang** [comment]
Very nice!

I'm partial to whichever style shows more title (because descriptions can be empty sometimes). 

However they seem to show the same content in which case for me **title text inside looks a lot better**, more streamlined for use as a menu item, more of a singular item/button that is visually distinct from other widget styles.

**Comment by jurialmunkey** [comment]
> I think the inside may be better... gives it a kind of "full package" look, whereas the outside makes the title and genre look isolated 

> Def inside text, Looks more "complete".

On its own, I agree, it looks great and more contained/complete; but it starts to look a bit weird when mixed in with other widgets where the text is outside because the style starts being a bit inconsistent.

> could lead to users asking for a way to switch the genre information off

On this point, I was thinking possibly a better approach is a global setting for all views. Then get rid of board because it would handle automatically. 

i.e. enable "inside" text makes Landscape turn into Board, and Apps turn into Landscape (but with text inside like Board). 

This would deal with the consistency issue a lot better because title text is either internal or external.

The only place I run into difficult is with Square because it doesn't neatly fit the same dimensions. It's always been a bit difficult - definitely the black sheep.


> I'm partial to whichever style shows more title (because descriptions can be empty sometimes).

Outside definitely shows more of the title. Inside loses 50 pixels of width in the title to accommodate the edge padding.

**Comment by L0bsang** [comment]
>On this point, I was thinking possibly a better approach is a global setting for all views. Then get rid of board because it would handle automatically.

This change initially didn't sound good to me because I mix board with text (next episodes) and landscape (recently aired episodes and certain discovery widgets) on my homepage so there is clear differentiation between them.
 
However after some thought I guess I could instead use Landscape for next episodes (would turn into board) and Apps for normal episodes and discovery widgets (would turn into landscape). 🤔 So probably ok, as long as the text is clearly visible when inside.

**Comment by jurialmunkey** [comment]
> So probably ok, as long as the text is quite legible when inside, and it looks good.

You can see what this looks like currently for Board with the option:
Skin Settings > Viewtypes > Show text labels on board widgets

The "Apps -> Landscape" switch would be the same as that but at the smaller size.

**Comment by xyzfre** [comment]
@jurialmunkey I noticed with this new feature A lot of things have labels under for example, in this picture every icon has the word folder ![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/49699869/a6d7a748-0641-4643-bd0d-d04537772fd6)

**Comment by CyberRooke** [comment]
> @jurialmunkey I noticed with this new feature A lot of things have labels under for example, in this picture every icon has the word folder ![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/49699869/a6d7a748-0641-4643-bd0d-d04537772fd6)

Yeah, and widgets and some files have 0.00 B

**Comment by L0bsang** [comment]
@xyzfre Is that the square style? 
Haven't been able to try the new version of the skin yet but I definitely agree that the repeated 'Folder' doesn't belong. 
Even in the mockup, "Trending lists" was repeated on every item but definitely not needed there, title label only is sufficient for non-content widgets.

**Comment by xyzfre** [comment]
Love the new large board mode but if there's a way to have an option for plugins categories back to the original smaller square size.

**Comment by JFG1000** [comment]
Yes jurial smaller plugin widgets would be great they are massive now

**Comment by jurialmunkey** [comment]
Smaller sizes require text underneath to fill the space. Either on or off. Not mix and match

**Comment by jurialmunkey** [comment]
Completely defeats the purpose of a setting designed to fill the space to the not do so. Turn off board mode if you want smaller sizes

**Comment by jurialmunkey** [comment]
All labels have been standardised. Suggest a good sublabel to use if you do not think folder is appropriate.

**Comment by CyberRooke** [comment]
> All labels have been standardised. Suggest a good sublabel to use if you do not think folder is appropriate. 

Is it possible to have a skin wide blank text as the sub label... it'll act the same way as a normal text would 🤔... just a suggestion: I don't really know the logistics

**Comment by L0bsang** [comment]
Happy to help sort out the appropriate sublabels as I definitely agree with others that it does apply some redundant information for non-content widgets (that honestly I think I could possibly live with, but will be trying it myself shortly).

What are our options? 
-Can sublabel visibility be content type dependent i.e. visible for shows and movies and episodes but not for folders and files?

-If sublabels always need to show for spacing reasons, what other labels can be shown there? For example instead of 0.00B for files we have a bit of their path or something like that, and then for folders in an addon the previous menu/folder is shown instead of type. So in @xyzfre's image it would show Movies (TMDB) and then once clicked it would be something like Trending (Movies) where the item in brackets is the sublabel, showing the previous folder. This may be tricky to implement.

-I imagine having the option to blank out labels/sublabels on a per-widget basis would also be a lot of work and not the right solution.

**Comment by L0bsang** [comment]
I've had a chance to try the changes and the new text widget works ok, but has too much repeated text when using a menu or set of folders with no plot information.

To illustrate:
New Text style
![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/162974281/8a738756-a3f6-4850-877d-68d4aa75f16d)
New Overview style
![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/162974281/8368e39d-5ce6-4b47-be66-cd6e99d9e6c7)
Icons
![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/162974281/b7c78f94-a5e8-4174-b1bc-8fada88c464e)
Apps
![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/162974281/0df691d8-01d5-4aa6-9430-2cd850a5ec14)

None of these are ideal for folders, closest to being suitable is text if it didn't have any labels.

**Comment by xyzfre** [comment]
@L0bsang thank you for posting these views already giving me some ideas 😁,I never tried using submenu widgets, but they look good, I've used them when I was using AH2.

**Comment by xyzfre** [comment]
> I've had a chance to try the changes and the new text widget works ok in certain cases but has repeated text when using it for a menu or set of folders with no plot information.
> 
> 
> 
> To illustrate, an example of how the various styles look when using a menu as a widget:
> 
> 
> 
> New Text style
> 
> ![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/162974281/1c15df45-519f-4d55-8558-a7a16d4ce7a1)
> 
> 
> 
> New Overview style
> 
> ![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/162974281/325b6b34-641d-4fe0-8372-9e705a2b2a73)
> 
> 
> 
> Icons
> 
> ![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/162974281/b02e0174-400e-4526-9fef-ca9b8699804f)
> 
> 
> 
> Apps
> 
> ![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/162974281/25405d5b-55af-4959-a6f0-eff61cb4be16)
> 
> 
> 
> Fully repeated information for the text/overview styles. Best display in terms of text shown and sizing seems to be text style without the redundant labels. 
> 
> Currently I am using apps/icon for these menu widgets and both seem better than text at the moment. Apps shows a bit more text than icon, but then uses landscape art, which some menus do not have.
> 
> 
> 
> But icon and apps styles have their own drawbacks. Here the 0.00B looks out of place next to my smart playlist items (some are playmedia and others are browse):
> 
> ![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/162974281/51107615-76ff-47d1-a138-48d2d91a7449)
> 
> Program addons have this 0.00B sublabel as well.
> 
> 
> 
> On the plus side, the trakt lists from TMDBHelper do look quite good with the new text style!
> 
> ![image](https://github.com/jurialmunkey/skin.arctic.fuse/assets/162974281/1411bfd4-131f-4104-bfde-eb873d439a26)
> 
> 
> 
> To summarise: I can learn to live with the new text style (decent for trakt lists) but it doesn't seem ideal for menus.

In your last pic with the trending popular list ,for it to show the text inside do I need to have the new setting the large board enabled can't remember exactly the name of the setting right now my apologies.

**Comment by jurialmunkey** [comment]
> In your last pic with the trending popular list ,for it to show the text inside do I need to have the new setting the large board enabled can't remember exactly the name of the setting right now my apologies.

Yes. That is what the setting does -- board widgets remove external widget text, then their bounding box is expanded to fill the space, and optionally widget text can then be placed inside the box.

**Comment by jurialmunkey** [comment]
@L0bsang 

> Can sublabel visibility be content type dependent i.e. visible for shows, movies and episodes but not for folders and files?

The big part of this push is to standardise the labels across all types, so no to changing visibility. However, *which* label is shown can be changed depending on content type -- e.g. this already happens with tvshows to show tvshow title instead of genre; or for live tv to show channel name.

> For example instead of 0.00B for files

I've removed size. Not even sure why I had that in the first place. There must be some place it is useful but I can't really remember why at the moment -- if I do remember I'll add it back with a more specific rule because it definitely shouldn't display constantly.

> then for folders in an addon the previous menu/folder is shown instead of type.

It does *sort of* do this already but is dependent on the plugin setting the ListItem.Property(widget) value. For e.g. in TMDbHelper if you go in TVDB > Awards you can see the label update. But this also feels a little redundant because if this is set, the label is already shown in the view line above the widget anyway.

> could have a bit of their path or something like that,

Possible but I tend to shy away from showing "technical" information like that as it is a bit meaningless to the layperson. It's meaningful for me but not necessarily someone else in my house wanting to watch a movie. 

It starts to feel like it is exposing the machinery too much if you get what I mean.

> I imagine having the option to blank out labels/sublabels on a per-widget basis would also be a lot of work and not the right solution. Potentially if sublabel is editable on a per-widget basis then users can put a blank text label where necessary as some have done for spotlight labels.

Less about the work and more that it isn't the path I want to head down. It's the opposite direction from standardisation of skin elements and makes the skin look inconsistent and unpolished.

I don't want to be adding in short circuits for users to try to "fix" the skin to their liking. I'm all for customisation but this isn't customisation -- it is trying to bend the skin design to be something else.

**Comment by L0bsang** [comment]
Fully agreed, thanks for the explanation of your direction. I especially agree about the path not being a good option for the label due to exposing the inner workings.

Looking forward to seeing how the adjustments will look. The most egregious sub-label indeed was the size, (which can be useful but not for the homepage widgets) so removing that is great.
